src/models/MSC3089Branch.ts

```diff
@@ -140,7 +140,14 @@ export class MSC3089Branch {
         const room = this.client.getRoom(this.roomId);
         if (!room) throw new Error("Unknown room");
 
-        const event = room.getUnfilteredTimelineSet().findEventById(this.id);
+        let event: MatrixEvent | undefined = room.getUnfilteredTimelineSet().findEventById(this.id);
+
+        // keep scrolling back if needed until we find the event or reach the start of the room:
+        while (!event && room.oldState.paginationToken) {
```

Comment:
do we specifically want `oldState`, or `currentState`? Using `oldState` is sufficiently rare where it should get a comment when we deliberately use it.

Reply:
Good spot. `currentState` makes more sense and I've also used the non-legacy implementation.